feat: module 101-02 content

bguiz opened the pull request:
- added a student guide for module 101-02
- added MCQs for module 101-02
- added a graded assessment for module 101-02

bguiz committed:
docs: add contents directory for module 101-02

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: create templates for the module materials

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: add empty template for module 101-02 student guide

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: write overview and learning outcomes for module 101-02

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: add code snippets and resources for module 101-02 student guide

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: add self assessment questions and answers to module 01-02 student guide

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: add template file for MCQs

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: added MCQs for module 101-02

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: add definitions and a few new resources in the module 101-02 student guide

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: add graded assessment for module 101-02

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

bguiz committed:
docs: add template for graded assessments

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>

michielmulders commented on hdc10102GradedAssessment/README.md:
````diff
@@ -0,0 +1,70 @@
+
+```shell
+mkdir -p ../hedera-dev-course-101-graded-assessments
+git cd hedera-dev-course-101-graded-assessments
````
Is this command correct? the `git cd`?

michielmulders commented on hdc10102GradedAssessment/script-hdc10102Task03.txt:
```diff
@@ -0,0 +1 @@
+https://
```
aren't we missing the hashscan url here for account 0.0.10?

bguiz replied:
This is intentional, this is the version that is incomplete, and the student is to replace it with the full URL. (so this serves as more of a hint)

michielmulders replied:
A/h I see!

michielmulders reviewed (commented):
thanks for adding the templates as well. I spotted the missing URL for hashscan grade assessment result and perhaps a typo in the command `git cd`?

bguiz committed:
docs: clean up graded assessment instructions

- includes PR review feedback from @michielmulders - https://github.com/hedera-dev/hedera-dev-course-101/pull/2#discussion_r1878142803

Signed-off-by: Brendan Graetz <bguiz@users.noreply.github.com>
